sycl/source/detail/pi_opencl.cpp

```diff
@@ -153,10 +155,11 @@ pi_program OCL(piProgramCreate)(pi_context context, const void *il,
           curPlatform, "clCreateProgramWithILKHR"));
 
   assert(funcPtr != nullptr);
-  resProgram = funcPtr(pi_cast<cl_context>(context), il, length,
-                         pi_cast<cl_int *>(err));
+    if (res_program != nullptr)
+      *res_program = pi_cast<pi_program>(funcPtr(pi_cast<cl_context>(context),
+                         il, length, pi_cast<cl_int *>(&err)));
```

Comment:
Please, apply clang-format to you patch.

Reply:
Done